benches/main.rs

```diff
@@ -33,12 +36,17 @@ pub fn bench_main(_c: &mut Criterion) {
         },
         Err(_) => panic!("Need to name a benchmark by the env var MMTK_BENCH"),
     }
+}
 
-    #[cfg(not(feature = "mock_test"))]
-    {
-        eprintln!("ERROR: Currently there are no benchmarks when the \"mock_test\" feature is not enabled.");
-        std::process::exit(1);
-    }
+pub fn bench_main(_c: &mut Criterion) {
+    // If the "mock_test" feature is enabled, we only run mock test.
+    #[cfg(feature = "mock_test")]
+    return bench_mock(_c);
+
+    // Some benchmarks rely on the "bench" feature to expose some private functions.
+    // Run them with `cargo bench --features bench`.
+    #[cfg(feature = "bench")]
+    bulk_meta::bzero_bset::bench(_c);
```

Comment:
Criterion benchmarks usually don't use the `match` statement to select benchmarks.  Instead it is similar to running tests.  It can be selected on the command line.  For example,

```
cargo bench --features test_private -- bzero
```

It will select all benchmarks that have "bzero" in the name.

The mock tests are special.  Because they rely on a global singleton `MockVM`, we can only run one test at a time.  Previously, we rely on the environment variable `MMTK_BENCH` to ensure only one mock test benchmark is selected at a time.

At the moment, I just isolated the `match` statement into the `bench_mock` function, and leave everything else in `bench_main`.  To install more benchmarks, we simply call `c.bench_function`, or call functions in other modules (such as `bulk_meta::bzero_bset::bench`) to install more benchmarks.  In the future, the `bench_main` function will simply call more functions, like these lines: https://github.com/mmtk/mmtk-core/pull/1174/files#diff-9ad1cee13898f4e5ac553eb567cd47a8de7d8852bf292c3d8d5991e23677db32R50-R51

In the future, we can refactor the `mock_bench` so that we can use the command line to select benchmarks, and as long as we select only one benchmark at a time, they will still be able to run.  But if that's impossible, we can still rely on the environment variable.

Reply:
I have now completely isolated mock benches from other benchmarks. Now regular benchmarks are in a separate module `regular_bench`, and each module inside it has its `fn bench` method for registering benchmarks.  When others add more regular benchmarks, they change the `fn bench` in the right module; when adding a mock benchmark, they edit `mock_bench::bench` which now holds the `match` statement.